scan-payment/src/main/java/com/getbouncer/scan/payment/Image.kt

```diff
@@ -241,6 +241,255 @@ fun Bitmap.zoom(
     return this.fragment(fromSegments, toSegments, toSize)
 }
 
+fun Bitmap.zoomOffset(
+    centerSize: Size,
+    previewSize: Size,
+    cardFinder: Rect,
+    toCenterDimension: Int,
+    toBorderWidth: Int
+): Bitmap {
+    require(cardFinder.left > 0 && cardFinder.right < previewSize.width && cardFinder.top > 0 && cardFinder.bottom < previewSize.height) {
+        "Card viewfinder window extends beyond preview boundary"
+    }
+    val newLeft = cardFinder.left * this.width / previewSize.height
+    val newTop = cardFinder.top * this.height / previewSize.height
+    val newRight = cardFinder.right * this.width / previewSize.width
+    val newBottom = cardFinder.bottom * this.height / previewSize.height
+
+    val scaledCardFinderRect = Rect(newLeft, newTop, newRight, newBottom)
+
+    val x = scaledCardFinderRect.centerX()
+    val y = scaledCardFinderRect.centerY()
+
+    require(x + centerSize.width < this.width && x - centerSize.width > 0 && y + centerSize.height < this.height && y - centerSize.height > 0) {
+        "Card viewfinder is to close to the edge of the screen"
+    }
+
+    require(toCenterDimension > 0 && toBorderWidth > 0) {
+        "Cannot create an image without a border or without a center"
+    }
+
+    val totalHeight = this.height - centerSize.height
+    val totalWidth = this.width - centerSize.width
+
+    val toSize = Size((2 * toBorderWidth + toCenterDimension), (2 * toBorderWidth + toCenterDimension))
+    var r1: Pair<Int, Int>
+    val r2: Pair<Int, Int>
+    val r3: Pair<Int, Int>
+    val r4: Pair<Int, Int>
+    var c1: Pair<Int, Int>
+    val c2: Pair<Int, Int>
+    val c3: Pair<Int, Int>
+    val c4: Pair<Int, Int>
+    var tr1: Pair<Int, Int>
+    val tr2: Pair<Int, Int>
+    val tr3: Pair<Int, Int>
+    val tr4: Pair<Int, Int>
+    var tc1: Pair<Int, Int>
+    val tc2: Pair<Int, Int>
+    val tc3: Pair<Int, Int>
+    val tc4: Pair<Int, Int>
```

Comment:
these names are hard to parse. is `r1 = right1`? is `r1 = row1`? Can you expand these out into full names? Also, do these represent sizes, points, etc? Let's use a class that describes what each of these are.